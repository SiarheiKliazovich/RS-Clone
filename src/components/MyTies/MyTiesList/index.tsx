import { FunctionComponent } from "react";
import { useDispatch, useSelector } from "react-redux";

import {
  myTieDeleted,
  myTiesFetching,
  myTiesFetched,
  myTiesFetchingError,
} from "../../../actions";
import { deleteTie, getTiesByUserId } from "../../../services/apiTies";
import { IAuthReducer } from "../../../interfaces/authReducer";
import { IMyTiesListComponent } from "../../../interfaces/myTiesListComponent";
import { useTranslation } from "react-i18next";

const MyTiesList: FunctionComponent<IMyTiesListComponent> = ({ ties }) => {
  const authStore = useSelector((state: IAuthReducer) => state.auth);
  const { user } = authStore;

  const dispatch = useDispatch();

  const { t } = useTranslation("dataLang");

  const deleteItem = async (id: string): Promise<void> => {
    try {
      await deleteTie(id);
      dispatch(myTieDeleted(id));
      dispatch(myTiesFetching());
      const ties = await getTiesByUserId(user._id);
      dispatch(myTiesFetched(ties));
    } catch {
      dispatch(myTiesFetchingError());
    }
  };

  return (
    <>
      <div className="my-ties-list__wrapper">
        {ties.map((item, i) => {
          return (
            <div key={i} className="my-ties-list__item">
              <div className="item__image">
                <img src={item.image} alt="" />
              </div>
              <div className="item__text">
                <div className="item__name">
                  {t("myTies.name")}: {item.name}
                </div>
                <div className="item__price">
                  {t("myTies.price")}: {item.price}$
                </div>
                <button
                  className="item__button"
                  onClick={() => deleteItem(item._id)}
                >
                  {t("myTies.delete")}
                </button>
              </div>
            </div>
          );
        })}
      </div>
    </>
  );
};

export default MyTiesList;
