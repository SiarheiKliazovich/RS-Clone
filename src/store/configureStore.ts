/* eslint-disable @typescript-eslint/ban-ts-comment */
import { applyMiddleware, createStore } from "redux";
import thunkMiddleware from "redux-thunk";

import loggerMiddleware from "./logger";
import rootReducer from "../reducers";

const configureStore = () => {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(rootReducer, middlewareEnhancer);

  // @ts-ignore:next-line
  if (process.env.NODE_ENV !== "production" && module.hot) {
    // @ts-ignore:next-line
    module.hot.accept("./../reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
};

export default configureStore;
