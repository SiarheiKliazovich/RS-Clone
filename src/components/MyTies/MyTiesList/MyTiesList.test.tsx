import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import MyTiesList from ".";

const mockStore = configureStore([]);

describe("MyTiesList", () => {
  // eslint-disable-next-line
  let store: any;

  beforeEach(() => {
    store = mockStore({
      auth: {
        user: {
          _id: "userId",
        },
      },
    });
  });

  test("should match snapshot", () => {
    const ties = [
      {
        _id: "1",
        name: "Test Tie 1",
        price: 10,
        image: "https://example.com/test1.jpg",
        userId: "user",
        sellerId: "seller",
      },
      {
        _id: "2",
        name: "Test Tie 2",
        price: 20,
        image: "https://example.com/test2.jpg",
        userId: "user",
        sellerId: "seller",
      },
    ];

    const { container } = render(
      <Provider store={store}>
        <MyTiesList ties={ties} />
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
