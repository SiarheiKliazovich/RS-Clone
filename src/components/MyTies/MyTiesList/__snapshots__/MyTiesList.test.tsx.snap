// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyTiesList renders correctly 1`] = `
<div
  class="market-block__products"
>
  <div
    class="row__products"
  >
    <div
      class="tie__product"
    >
      <div
        class="tie__products_img"
      >
        <img
          alt=""
          class="products_img"
          src="https://example.com/test1.jpg"
        />
      </div>
      <div
        class="tie__products_discription"
      >
        <div
          class="tie__products_name"
        >
          myTies.name
          : 
          Test Tie 1
        </div>
        <div
          class="tie__products_price"
        >
          myTies.price
          : 
          10
          $
        </div>
        <div
          class="tie__products_wrapper"
        >
          <button
            class="btn__like"
          >
            <i
              class="fa fa-remove"
            />
            myTies.delete
          </button>
        </div>
      </div>
    </div>
    <div
      class="tie__product"
    >
      <div
        class="tie__products_img"
      >
        <img
          alt=""
          class="products_img"
          src="https://example.com/test2.jpg"
        />
      </div>
      <div
        class="tie__products_discription"
      >
        <div
          class="tie__products_name"
        >
          myTies.name
          : 
          Test Tie 2
        </div>
        <div
          class="tie__products_price"
        >
          myTies.price
          : 
          20
          $
        </div>
        <div
          class="tie__products_wrapper"
        >
          <button
            class="btn__like"
          >
            <i
              class="fa fa-remove"
            />
            myTies.delete
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
